code/game/machinery/autolathe.dm

```diff
@@ -615,11 +615,13 @@
 	queue = Q
 
 
+//Autolathes can eject decimal quantities of material as a shard
 /obj/machinery/autolathe/proc/eject(var/material, var/amount)
 	if(!(material in stored_material))
 		return
 
-	amount = round(amount) //The autolathe can contain less than a full sheet, but only ejects whole sheets
+	var/whole_amount = round(amount)
+	var/remainder = amount - whole_amount
```

Comment:
It's better to start actions after all check.
Especially "do actions with `amount`" before "check `amount` value"